### ISSUE TITLE
[steamshovel] option for no cleanup on artist setting change (Trac #1081)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1075]
When changing a setting on an artist (specifically LEDPowerHouse) I'd like to not run the full cleanup, but to just change the setting and do another draw.  Is there a way to make this happen?

Migrated from https://code.icecube.wisc.edu/ticket/1081
```json
{
    "status": "closed", 
    "changetime": "2016-03-18T21:14:10", 
    "description": "When changing a setting on an artist (specifically LEDPowerHouse) I'd like to not run the full cleanup, but to just change the setting and do another draw.  Is there a way to make this happen?", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1458335650323600", 
    "component": "combo core", 
    "summary": "[steamshovel] option for no cleanup on artist setting change", 
    "priority": "major", 
    "keywords": "", 
    "time": "2015-07-30T16:51:29", 
    "milestone": "", 
    "owner": "hdembinski", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150801T12:51:23: hdembinski commented:

> I suppose that steamshovel is delete'ing and new'ing artists more often than necessary, but I did not notice this. I can try to look into it, but could you have a look as well?

**Comment by icecube-trac** [comment]
Trac update at 20150801T13:53:32: david.schultz commented:

> When debugging LDEPowerHouse, I saw that it would new/delete about 15-20 times every time you switched frames.  While it doesn't take long, it made lots of logging for me :(

**Comment by icecube-trac** [comment]
Trac update at 20150803T09:23:08: hdembinski commented:

> i am also not happy about all those new/delete every time you switch frames, but it looks like a deep design decision, difficult to change. artists are dynamically created by ArtistFactories. if you look into Scenario.cpp, you can see that the factory often just creates a temporary artist to use its "isValidKey" function.

**Comment by icecube-trac** [comment]
Trac update at 20150803T09:43:32: 
* hdembinski changed status from "new" to "closed"
* hdembinski commented:

> In changeset 135789:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="135789"
> phasing out ArtistFactories, we do not want to new/delete Artists so often anymore, closes #1075
> ```


* hdembinski changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20160318T21:14:10: 
* olivas changed milestone from "offline V15-12-00" to ""
* olivas commented:

> Milestone offline V15-12-00 deleted
